tests/saitests/sai_qos_tests.py

```diff
@@ -2119,48 +2119,64 @@ def runTest(self):
                 if actual_pkts_num_leak_out > pkts_num_leak_out:
                     send_packet(self, src_port_id, pkt, actual_pkts_num_leak_out - pkts_num_leak_out) 
 
-            q_wm_res, pg_shared_wm_res, pg_headroom_wm_res = sai_thrift_read_port_watermarks(self.client, port_list[dst_port_id])
-            print >> sys.stderr, "Init pkts num sent: %d, min: %d, actual watermark value to start: %d" % ((pkts_num_leak_out + pkts_num_fill_min), pkts_num_fill_min, q_wm_res[queue])
-            if pkts_num_fill_min:
-                assert(q_wm_res[queue] == 0)
+            if asic_type == 'cisco-8000':
+                pkt_count_list = [50,1000,4000,7000,12500]
+                q_wm_res, pg_shared_wm_res, pg_headroom_wm_res = sai_thrift_read_port_watermarks(self.client, port_list[dst_port_id])
+                q_wm_res_base = q_wm_res[queue]
+                for pkt_cnt in pkt_count_list:
+                    q_wm_res_base = q_wm_res[queue]
+                    sai_thrift_port_tx_disable(self.client, asic_type, [dst_port_id])
+                    send_packet(self, src_port_id, pkt, pkt_cnt)
+                    sai_thrift_port_tx_enable(self.client, asic_type, [dst_port_id])
+                    time.sleep(8)
+                    q_wm_res, pg_shared_wm_res, pg_headroom_wm_res = sai_thrift_read_port_watermarks(self.client, port_list[dst_port_id])
+                    # verify if new wm value is greater than or equal to base wm value
+                    # Also verify that new wm value is greater than or equal to the number of packets sent
+                    assert(q_wm_res[queue] >= q_wm_res_base)
```

Comment:
Looks like Cisco implements a totally different logic here, in the previous test case, it will send chunk of pkts, and check the watermark is in the expected range.
But the new logic here is just to check the watermark is increased? If we send 1000 pkts, what's the expected watermark value?